pfcpiface/p4rt_translator.go

```diff
@@ -404,6 +502,61 @@ func (t *P4rtTranslator) ParseAccessIPFromReadInterfaceTableResponse(resp *p4.Re
 	return nil, ErrOperationFailed("parse Access IP from P4Runtime response")
 }
 
+func (t *P4rtTranslator) BuildApplicationsTableEntry(pdr pdr, internalAppID uint8) (*p4.TableEntry, error) {
+	applicationsBuilderLog := log.WithFields(log.Fields{
+		"pdr": pdr,
+	})
+	applicationsBuilderLog.Trace("Building P4rt table entry for applications table")
+
+	tableID := t.tableID(TableApplications)
+	entry := &p4.TableEntry{
+		TableId: tableID,
+		// priority for UP4 cannot be greater than 65535
```

Comment:
However, priority in ROC is from 0 (low) to 250 (high). We need to understand how precedence is calculated to do our assumption here.

Reply:
this is the outcome from STC testing. The UP4 app rejected table entry with the following message: `Priority cannot be greater than 65535`. I see that `MAX_PRIORITY` = 65535 is used among different ONOS modules: https://github.com/opennetworkinglab/onos/blob/2b4de873e4033b7973b399d25cb8828a73bc2e24/core/api/src/main/java/org/onosproject/net/flow/FlowRule.java#L32. May it be the reason?

Reply:
Yes, definitely that's the reason, thanks for the explanation!

Reply:
It would be safer to check if precedence < 65535 and return an error if not (maybe in an utility function shared with other translation functions, since we use ternary in different tables). Limiting the table entry priority to 250 is too conservative without any benefit. I would consider only the ONOS limit for now.

Reply:
@ccascone I'm going to merge this as is and add the precedence verification in a separate PR.